project/paths.ts

```diff
@@ -40,6 +44,9 @@ export class Paths {
 
   readonly gdscriptPath: string
 
+  additionalIgnores: string[]
```

Comment:
Why do we need to special case tsFileIgnores vs additionalIgnores? Can we just have a single list of ignored files that gets appended to our ignoredPaths() list?

Reply:
That is due how the `!` functionality works with anymatch with files (afaict).
We exclude all files. 
But then !include ts files which are not d.ts.
Then ignoring specific ts files will be overwritten by the !include.
Thus we need to not !include the singular ts files with our !*.ts rule.
So the user rules get split up in two.